### PR TITLE
fix: The `prefixCls` feature of `Modal.info` method is incomplete #12546

### PR DESCRIPTION
**Pull request by concefly**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for branch `feature`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

------

https://github.com/ant-design/ant-design/issues/12546

Add an extra `modalPrefixCls` property to assign the prefixCls of Modal container.

### CODE REVIEW

**Comment by zombieJ** [comment]
ref: #10855

**Comment by zombieJ** [comment]
感觉以前设计这个的 `prefixCls` 不太好，它只负责 Modal content 的 `prefixCls`。要修改 Modal 就需要额外搞一个 `modalPrefixCls` 属性，不直觉不过也好像没什么办法。
@afc163 @yesmeck 感觉呢？这个 prefixCls 要不要调整？

**Comment by afc163** [comment]
要不然 prefixCls 改回 `ant-modal` 然后加一个 className `ant-confirm` 做 confirm 等的自定义样式？

**Comment by zombieJ** [comment]
> @afc163 
> 要不然 prefixCls 改回 ant-modal 然后加一个 className ant-confirm 做 confirm 等的自定义样式？

如果是这样，和以前就不太兼容了，放 feature branch 吧。此外 `ant-confirm` 也可以一并调整成 `${prefixCls}-confirm`了。

**Comment by zombieJ** [comment]
@concefly , 之前也考虑了这个，但是这种用法就和其他的 prefixCls 不一样了。所以才想以 Modal 为主这样。

File: components/modal/confirm.tsx [hunk 1]
```diff
@@ -18,7 +18,8 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
   const { onCancel, onOk, close, zIndex, afterClose, visible, keyboard, centered, getContainer, okButtonProps, cancelButtonProps } = props;
   const iconType = props.iconType || 'question-circle';
   const okType = props.okType || 'primary';
-  const prefixCls = props.prefixCls || 'ant-confirm';
+  const modalPrefixCls = props.modalPrefixCls || 'ant-modal';
+  const contentPrefixCls = props.prefixCls || 'ant-confirm';
```
Comment (afc163) [inline components/modal/confirm.tsx]: contentPrefixCls 改名为 confirmPrefixCls 吧。

**Comment by concefly** [comment]
所以这个 RP 能合吗，还是要调整

**Comment by zombieJ** [comment]
@concefly ,改成 prefixCls 对应 ant-modal 吧， ant-confirm 先保留不让用户自己改这样

**Comment by concefly** [comment]
@zombieJ 

> ant-confirm 先保留不让用户自己改这样

这意味着 js 硬编码 ant-confirm 类名，不能通过 prefixCls 设置 ${prefixCls}-confirm 类名了吗？但现在 less 里面 ant-confirm 类名是可以自定义前缀的。这么改，js 和 less 的 prefixCls 处理就不一致了。

**Comment by zombieJ** [comment]
那 `ant-confirm` 调整成 `ant-modal-confirm` 如何？
```jsx
`${prefix}` => 'ant-modal' (default)
`${prefix}-confirm` => 'ant-modal-confirm' (default)
```

@concefly @afc163  @yesmeck

**Comment by afc163** [comment]
👍 这样更准确。

**Comment by afc163** [comment]
image

绑定下账号。

**Pull request merged**